### PR TITLE
Add OpenWeatherMap API Tool

### PR DESCRIPTION
**Pull request by Honkware**
Added tool for OpenWeatherMap API

### CODE REVIEW

File: langchain/utilities/openweathermap.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+"""Util that calls OpenWeatherMap using PyOWM."""
+
+from typing import Any, Dict, Optional
+
+from pydantic import Extra, root_validator
+from pyowm import OWM
```
Comment (hwchase17) [inline langchain/utilities/openweathermap.py:6]: lets not add this in the file, but treat as optional dependency so only do inside functions when needed

File: langchain/__init__.py [hunk 2]
```diff
@@ -51,6 +51,7 @@
 from langchain.utilities.serpapi import SerpAPIWrapper
 from langchain.utilities.wikipedia import WikipediaAPIWrapper
 from langchain.utilities.wolfram_alpha import WolframAlphaAPIWrapper
+from langchain.utilities.openweathermap import OpenWeatherMapAPIWrapper
```
Comment (hwchase17) [inline langchain/__init__.py:54]: lets not add here, trying to discourage imports directly from root

**Review by hwchase17 — COMMENTED** [review]
thanks for this + the example! some small changes

File: langchain/utilities/openweathermap.py (hunk 1 above)
Comment (Honkware) [reply langchain/utilities/openweathermap.py]: I removed the import from the file and added it as an optional dependency

**Review by hwchase17 — APPROVED** [review]
thanks!

**Pull request merged**